### PR TITLE
Add an update-codegen to regenerate the static redirects.

### PR DESCRIPTION
**Pull request by evankanderson**
I also noticed a bug in redir-gen that we weren't going through list pagination, and so missed several repos.

/assign @mattmoor 

I think adding `./hack/update-codegen.sh` will cause nightly PRs to update when needed, but won't block other PRs with requiring this to be regenerated. Is that correct?

### CODE REVIEW

**Comment by mattmoor** [comment]
@evankanderson We probably need to add `knative/website` to `repos.yaml` in `knative-sandbox/knobots` for the daily PRs, I don't think it ever made the list.

For the `verify-codegen` presubmit check, the action sync should take care of that (once the actions land here) 🤞

File: tools/redir-gen/main.go [hunk 1]
```diff
@@ -67,15 +67,27 @@ func fetchRepos(orgs []string) ([]repoInfo, error) {
...
+				return nil, err
+			}
+			for _, r := range repos {
+				if !allowedRepoRe.MatchString(*r.Name) {
+					log.Printf("Ignoring repo %s, matched by ignore %q", *r.Name, allowedRepoRe)
+					continue
+				}
+				if *r.Archived {
+					log.Print("Ignoring archived repo: ", *r.Name)
+				}
```
Comment (mattmoor) [inline tools/redir-gen/main.go:85]: Hmm, why did Build show up 🤔

Comment (mattmoor) [reply tools/redir-gen/main.go:85]: Ha!  no `continue`

Comment (evankanderson) [reply tools/redir-gen/main.go:85]: D'oh, that'll teach me to refector without tests.

**Comment by evankanderson** [comment]
I'm not sure we want a `verify` for these redirects, because they vary based on external factors (what repos have been created), and I'm not sure that we want to block PR submits because a new repo got created somewhere. Given that, I think just an `update-codegen.sh` is appropriate.

**Review by mattmoor — APPROVED** [review]
/lgtm

**Pull request merged**